.github/workflows/ci.yml

```diff
@@ -44,33 +43,55 @@ jobs:
           dotnet tool restore
           dotnet tool run dafny-reportgenerator summarize-csv-results --max-resource-count 10000000 . || true
 
+  build_and_test_lean:
+    name: Build and Test Lean
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+    steps:
+      - name: Checkout cedar-spec
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install Lean
+        shell: bash
+        run: |
+              wget https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh
+              bash elan-init.sh -y
+      - name: Build
+        working-directory: ./cedar-lean
+        shell: bash
+        run: source ~/.profile && lake build Cedar
+      - name: Test
+        working-directory: ./cedar-lean
+        shell: bash
+        run: source ~/.profile && lake exe CedarUnitTests
+
   build_and_test_drt:
     name: Build and Test DRT
     runs-on: ubuntu-latest
     strategy:
       matrix:
         toolchain:
           - stable
-
     steps:
       - name: Checkout cedar-spec
         uses: actions/checkout@v3
         with:
           submodules: recursive
+      - name: Install Lean
+        shell: bash
+        run: |
+              wget https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh
+              bash elan-init.sh -y
       - name: Get Java 17
         uses: actions/setup-java@v3
         with:
           distribution: 'corretto'
           java-version: '17'
           cache: 'gradle'
-      - name: Build Dafny def engine
-        working-directory: ./cedar-dafny
-        shell: bash
-        run: make compile-difftest
-      - name: Build cedar-dafny-java-wrapper
-        working-directory: ./cedar-dafny-java-wrapper
-        shell: bash
-        run: ./gradlew build dumpClasspath
```

Comment:
just checking, are we removing the Dafny DRT from our CI entirely?  or do we want to keep both in CI for a few weeks?

Reply:
I was planning on keeping it in the CI until the Dafny formalization is "fully" deleted, which should be a few weeks at least.

Reply:
Looking at this again: I didn't read your question properly. The Dafny build steps are included in the `build.sh` script, which this PR updates the CI to use. (Before, the `build.sh` script and the CI were doing something similar, but were not identical.)